### PR TITLE
THE ARCHITECT UPDATE: Craftable Guns by Engineers, Keep Architect, Blunderbusses, and more!

### PR DESCRIPTION
**Pull request by Slorbin**
## About The Pull Request

**Adds a Blunderbuss!**

It does less damage than the Arquebus from anywhere but point-blank, and it has pretty high range, but it does something!

https://github.com/user-attachments/assets/9a3974c3-4662-48b3-8ecf-9ee8d1f9c3a5

**Adds a Clockwork Drill!**

It's a one-of-a-kind item unique to the Architect role that lets you break down walls much faster! It's one-of-a-kind and spawns with the architect, who is keep-aligned, so it's used for remodeling without having to sledgehammer a wall for 20 minutes straight! It's noisy as all hell so everyone will know the moment you even start wielding it.

https://github.com/user-attachments/assets/acd322c7-dc14-46ca-9773-9fed6aa22b9d

**Replaces the Archivist with the Architect!**

Originally it was going to be its own role, but community members suggested replacing the Archivist since the Archivist is currently 'boring as hell', and the Architect also fills the role of 'smart guy in keep that isn't a magician/doctor'. He still has the same duties as the Archivist when it comes to managing the archives, which let's be honest no one ever does, but his role no longer sucks so much ass because he has a secondary job: Being the keep's craftsman, helping with repairs, creation, and remodeling of the fortress. He spawns with the aforementioned drill, and the blunderbuss, and gets 2 gun skill.

**Guns, gun parts, and ammo are now craftable!**

Guns can be crafted by those with a 4 in engineering!
Currently, this is only artificers and architects unless you roll legendary mason. Also, Artificers get 2 gun skill now, since they're smart guys who aren't going to blow themselves up when trying to use machinery. They're engineers, for Astrata's sake!

It requires you to make a gun stock, a gun lock, and a gun barrel, and then combine them to make either an arquebus or blunderbuss. Also, anyone can saw off an arquebus to make a handgonne, since it's just a worse short arquebus. You still need to wield it in two hands for some reason even. Runelock and the arquebus pistol are not craftable as they are special items.

You can also make a powderflask from the same stuff needed to make a waterflask plus syrum of ash, replacement ramrods for 1 iron per 3, and musketballs/ammunition from iron ingots.

## Why It's Good For The Game

You can't make replacement ramrods, powderflasks or musketballs, which is a huge oversight.

You can now make more guns, but they're pretty rare, only from an artificer who is willing to put a fair amount of work in!

The Archivist was a boring as fuck role who was only played by people with a weird 40-year-old-virgin fetish so now it doesn't suck so hard and is actually fun to play.

You can ACTUALLY remodel the keep now without it taking a whole HOUR to knock down ONE set of walls.

Blunderbusses are COOL/

### CODE REVIEW

**Comment by Elli-Skala** [comment]
level three smithing and climbing
level four carpenter and masonry 
level SIX engineering

File: code/game/objects/items/rogueweapons/ranged/gunpowder/blunderbuss.dm [hunk 1]
```diff
@@ -0,0 +1,276 @@
...
+				"sound/arquebus/arquefire4.ogg", "sound/arquebus/arquefire5.ogg")
+	. = ..()
+
+/obj/item/gun/ballistic/blunderbuss/attack_right(mob/user)
+	if(user.get_active_held_item())
+		return
+	else
+		if(myrod)
+			playsound(src, "sound/items/sharpen_short1.ogg",  100)
+			to_chat(user, "<span class='warning'>I draw the ramrod from the [src]!</span>")
```
Comment (Gyrandola) [inline code/game/objects/items/rogueweapons/ranged/gunpowder/blunderbuss.dm:66]: Please spannify this with the span_warning helper. This goes for all the ones that follow.

File: code/modules/jobs/job_types/roguetown/yeomen/architect.dm [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+	if(H.mind)
+		H.mind.adjust_skillrank(/datum/skill/labor/mining, 3, TRUE)
+		H.mind.adjust_skillrank(/datum/skill/combat/maces, 2, TRUE)
+		H.mind.adjust_skillrank(/datum/skill/combat/firearms, 2, TRUE)
+		H.mind.adjust_skillrank(/datum/skill/combat/wrestling, 2, TRUE)
+		H.mind.adjust_skillrank(/datum/skill/combat/unarmed, 2, TRUE)
+		H.mind.adjust_skillrank(/datum/skill/craft/carpentry, 4, TRUE)
+		H.mind.adjust_skillrank(/datum/skill/craft/masonry, 4, TRUE)
+		H.mind.adjust_skillrank(/datum/skill/craft/crafting, 5, TRUE)
+		H.mind.adjust_skillrank(/datum/skill/craft/engineering, 6, TRUE)
```
Comment (Gyrandola) [inline code/modules/jobs/job_types/roguetown/yeomen/architect.dm:42]: Due to all other skills he gets, his role in the keep, his gun, etc.. he should not be an artificer but better. Same engineering level at most so we dont make artificer redundant.

Also generally no one should ever start with legendary skills, veteran being the exception due to the debuffs.

File: code/rt.dm [hunk 3]
```diff
@@ -1,5 +1,5 @@
 #ifndef TESTING
-//    #define FASTLOAD
+    #define FASTLOAD
```
Comment (Gyrandola) [inline code/rt.dm:2]: Be careful

**Review by Gyrandola — CHANGES_REQUESTED** [review]

**Comment by Gyrandola** [comment]
Im okay with test merging this once the review has been addressed.

The whole gun/ammo making process might have to be changed if it proves too easy.

**Comment by DeerJesus** [comment]
> the drill is immersion breaking could it at least have the sound changed so its not a gas engine drill, nothing about this seems renaissance nor fantasy

Yeah I was just reusing the chainsaw sound... I think I'll look around old clock cult code for a more clockwork kinda sound? I'll do that along with the other changes when I get to it

**Comment by Gyrandola** [comment]
Okay yeah, I think carl makes a good point.

Architect could replace the now defunct mason instead of the archivist, which still fills a niche.

The keep having its own smith is also not a good idea.

**Comment by DeerJesus** [comment]
Wait I thought that the Artificer replaced the mason?

**Comment by DeerJesus** [comment]
> Okay yeah, I think carl makes a good point.
> 
> Architect could replace the now defunct mason instead of the archivist, which still fills a niche.
> 
> The keep having its own smith is also not a good idea.

So you want this to be part of the artificer's guild and NOT part of the keep?

**Comment by Gyrandola** [comment]
> > Okay yeah, I think carl makes a good point.
> > Architect could replace the now defunct mason instead of the archivist, which still fills a niche.
> > The keep having its own smith is also not a good idea.
> 
> So you want this to be part of the artificer's guild and NOT part of the keep?

that might be for the best honestly. We want towner roles to become more and more appealing and full of content.

**Comment by Coolgat3** [comment]
whatever you do just don't give bandits guns they're already op enough as is being able to 4v15, thanks.

**Comment by DeerJesus** [comment]
> whatever you do just don't give bandits guns they're already op enough as is being able to 4v15, thanks.

God no lol I'm not giving bandits guns. If bandits somehow get guns from them being craftable it means something has gone SERIOUSLY wrong in game and the other side has been ROYALLY fucked. And they don't even have firearms skill- so even if they steal one they'd have to waste so much ammo just to get to the min level to use them.

In any case I might not be able to work on this PR for a little. Thanksgiving and college stuff. I'll slowly chop away at the changes.

**Comment by DeerJesus** [comment]
the files are working but for some reasoin it wont let me mark the conflict as resolved so IDFK

**Review by Gyrandola — APPROVED** [review]

**Comment by wow50000** [comment]
could you remove the smithing from them so towner smiths arent just there to kill theyself ?

**Comment by Elli-Skala** [comment]
after playing a round as architech i feel abit lost at its purpose
if it was still a keep role some things would make sense such as smithing or engineering but currently it just out shines the artificers while also taking skills away from the guild smith
its weirdly armed to the teeth as if it was another member of the garrison unlike any other yeoman. 
i think the smithing should be removed, and engineering lowered to just 2 so they can build engineering buildings only
if they are sticking to being a yeoman I think the blunder buss should be replaced with a satchle/backpack and a wood axe 
how ever I find rather you are building for the keep, or bugging off to no where and building for your self. which I start to think that maybe staying as the keep's handy man may have been better off

**Comment by DeerJesus** [comment]
> after playing a round as architech i feel abit lost at its purpose if it was still a keep role some things would make sense such as smithing or engineering but currently it just out shines the artificers while also taking skills away from the guild smith its weirdly armed to the teeth as if it was another member of the garrison unlike any other yeoman. i think the smithing should be removed, and engineering lowered to just 2 so they can build engineering buildings only if they are sticking to being a yeoman I think the blunder buss should be replaced with a satchle/backpack and a wood axe how ever I find rather you are building for the keep, or bugging off to no where and building for your self. which I start to think that maybe staying as the keep's handy man may have been better off

yeah the arming thing is due to again it supposed to be akeep member

i am still petitioning for this to be a keep thing gyran

**Comment by Frienderanian** [comment]
This role is genuinely so broken and awful, I want whatever crack the dude who made this was SMOKING. I got home from 14 hours of work and decided to play a single round of Ratwood, saw that "Architect" was a new role, saw this PR without reading the comments and picked it. God what a mistake, man.

You're basically just an Artifacer (an already STUPIDLY BROKEN class that shouldn't be in the game) except now you start with a drill that can bust down any wall in 1 second, a gun and EXPERT in five different skills and FOUR stats above 12. 
Screenshot 2024-12-02 234518 These were the skills and stats I got, ignore the Legendary Engineering, that was after I leveled it up twice. 

Best part? YOU'RE NOT EVEN ALIGNED WITH THE KEEP!!! I picked the role because I thought it replaced Archivist and I wanted to help out around the Keep and be useful, but nah, I just get to bum around the town and use the Drill to destroy the [ugly fucking dogshit MASSIVE Stewards Building](https://github.com/Rotwood-Vale/Ratwood-Keep/pull/1764) that raped the map design of the Town Square and literally nothing stops me from doing this. 

So, I dug into the Stewards tower, extorted him for money to repair the walls I broke then bummed around inside the Artifacer's house (the only house I had Keys for) and duped Steel like "a Cheating little faggot" for the next two hours and logged off after a Knight picked a 1v1 with a Bandit and somehow lost despite me unloading two fucking shotgun blasts into the bandits fucking leg. 

Fuck Artifacer, fuck this role.

**Comment by RyszardParys** [comment]
Craftable guns will never be fine...
I will forever hate the "click to win the fight" stick, and making them craftable in ANY form is coal.

**Comment by Gyrandola** [comment]
Give him the mason's old skills and it should be fine.

As for the drill being strong, its a unique item that greatly empowers a towner role, thats what we want.

**Comment by Gyrandola** [comment]
> not a fan of it just being an objectively better towner smith with more skills, better stats and a gun and a drill...

yeah he should be good at building, not smithing or engineering

**Comment by DeerJesus** [comment]
> > not a fan of it just being an objectively better towner smith with more skills, better stats and a gun and a drill...
> 
> 
> 
> yeah he should be good at building, not smithing or engineering

please let me put him back in the keep dawg

**Comment by Frienderanian** [comment]
I'm sorry if I was a little harsh with my criticisms earlier, I get that y'all are just trying to make a fun game and I understand the arguments behind not having ANOTHER keep role. I genuinely did not realize how horrifically game breaking Artifacer as a class is until I picked Architect and was forced to hang out with an Artifacer and learn the mechanics. The world of Ratwood Keep is a genuine Garden of Eden fucking PARADISE, the people of Rockhill live in a post scarcity economy where Healing is free and easy, food is plentiful and even precious metals are infinitely duplicatable. When I get home from another 14 hour shift, I'm picking Architect and I am duping guns and arming the entire populace. Thank you, Ratwood Keep.

**Comment by Gyrandola** [comment]
> I'm sorry if I was a little harsh with my criticisms earlier, I get that y'all are just trying to make a fun game and I understand the arguments behind not having ANOTHER keep role. I genuinely did not realize how horrifically game breaking Artifacer as a class is until I picked Architect and was forced to hang out with an Artifacer and learn the mechanics. The world of Ratwood Keep is a genuine Garden of Eden fucking PARADISE, the people of Rockhill live in a post scarcity economy where Healing is free and easy, food is plentiful and even precious metals are infinitely duplicatable. When I get home from another 14 hour shift, I'm picking Architect and I am duping guns and arming the entire populace. Thank you, Ratwood Keep.

>I'm sorry
>proceeds to be a complete shithead

**Comment by Frienderanian** [comment]
> > I'm sorry if I was a little harsh with my criticisms earlier, I get that y'all are just trying to make a fun game and I understand the arguments behind not having ANOTHER keep role. I genuinely did not realize how horrifically game breaking Artifacer as a class is until I picked Architect and was forced to hang out with an Artifacer and learn the mechanics. The world of Ratwood Keep is a genuine Garden of Eden fucking PARADISE, the people of Rockhill live in a post scarcity economy where Healing is free and easy, food is plentiful and even precious metals are infinitely duplicatable. When I get home from another 14 hour shift, I'm picking Architect and I am duping guns and arming the entire populace. Thank you, Ratwood Keep.
> 
> ->I'm sorry ->proceeds to be a complete shithead

Please present an argument for why I am wrong, if you do not have an argument for why I'm wrong, just say so and that's perfectly fine too.

**Pull request merged**